### PR TITLE
Fix: Fix UI error in gcode line review

### PR DESCRIPTION
**Pull request by bbaa77770**

### CODE REVIEW

File: src/app/flux/printing/index.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ import {
     RIGHT_EXTRUDER,
     LEFT_EXTRUDER_MAP_NUMBER,
     RIGHT_EXTRUDER_MAP_NUMBER,
-    DUAL_EXTRUDER_TOOLHEAD_FOR_SM2
+    DUAL_EXTRUDER_TOOLHEAD_FOR_SM2,
+    ALIGN_OPERATION
```
Comment (MaShizhen) [inline src/app/flux/printing/index.js:22]: > ```ALIGN_OPERATION``` 

This variable is often changed recently. It is better to rebase the master

File: src/app/styles/global-edge-distance.styl [hunk 2]
```diff
@@ -64,6 +67,9 @@
 .margin-left-24 {
  margin-left: $px-grade-12;
 }
+.margin-left-n-8 {
+ margin-left: $px-grade-n-4;
```
Comment (MaShizhen) [inline src/app/styles/global-edge-distance.styl:71]: Indent should be unified

File: src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx [hunk 3]
```diff
@@ -311,82 +309,86 @@ function VisualizerPreviewControl() {
...
+                                            return (
+                                                <PreviewType
+                                                    fatherContent={fatherContent}
+                                                    fatherColor={fatherColor}
+                                                    isDropdown={isDualExtruder}
+                                                    childrenObjects={
+                                                        isDualExtruder ? [
+                                                            {
+                                                                value: allShowTypes[LEFT_EXTRUDER][showType],
+                                                                content: i18n._('key-Printing/Preview-Tool0'),
```
Comment (MaShizhen) [inline src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx:361]: It is suggested to change to `'key-Printing/Preview-Extruder L'`

File: src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx [hunk 4]
```diff
@@ -208,7 +163,50 @@ function VisualizerPreviewControl() {
     }
 
     // TODO
-
+    const lineTypeObjects0 = [
```
Comment (MaShizhen) [inline src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx:166]: ExtruderLineType ?
structureLineType ?

File: src/app/ui/components/ColorSelector/ColorSelectorWrapper.jsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+        if (resetRecentColors) {
+            let newRecentColors = recentColors;
+            const index = newRecentColors.indexOf(color);
+            if (index > -1) {
+                newRecentColors.splice(index, 1);
+            }
+            newRecentColors = [color, ...recentColors];
+            newRecentColors.splice(10, 1);
+            localStorage.setItem(`color-selector-${recentColorKey}`, newRecentColors);
```
Comment (jane-rose) [inline src/app/ui/components/ColorSelector/ColorSelectorWrapper.jsx:27]: use `storeManager` not `localStorage`

File: src/app/ui/views/ProfileManager/SettingItem.jsx [hunk 6]
```diff
@@ -220,17 +218,16 @@ function SettingItem({ definitionKey, settings, isDefaultDefinition = () => true
...
-                <div className={classNames(styles['color-selection'])}>
-                    <span>
-                        color
-                    </span>
-                    <GithubPicker
-                        width={275}
-                        triangle="hide"
+                <div>
+                    <ColorSelector
+                        recentColorKey="profile-manager"
```
Comment (jane-rose) [inline src/app/ui/views/ProfileManager/SettingItem.jsx:223]: constant

File: src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx (hunk 4 above)
Comment (jane-rose) [reply src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx:166]: +1，change name

File: src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx [hunk 7]
```diff
@@ -311,82 +309,86 @@ function VisualizerPreviewControl() {
...
+                                                                value: allShowTypes[LEFT_EXTRUDER][showType],
+                                                                content: i18n._('key-Printing/Preview-Tool0'),
+                                                                onChangeValue: togglePreviewOptionFactoryByTypeAndDirection(showType, showTypeName, LEFT_EXTRUDER)
+                                                            }
+                                                        ]
+                                                    }
+                                                />
+                                            );
+                                        }))}
+                                        {renderLineType && (lineTypeObjects1.map((obj) => {
```
Comment (xiekun1992) [inline src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx:380]: use a good name for `lineTypeObjects1` and `lineTypeObjects0`

**Review by xiekun1992 — CHANGES_REQUESTED** [review]

File: src/app/store/local-storage.js [hunk 8]
```diff
@@ -80,16 +80,19 @@ const getLocalStore = (name) => {
...
     widgetStore,
     machineStore,
+    componentsStore,
     clear: () => {
         machineStore.clear();
         widgetStore.clear();
+        componentsStore.clear();
     },
     get: () => {
-        return _.merge(machineStore.get(), widgetStore.get());
```
Comment (jane-rose) [inline src/app/store/local-storage.js:92]: missing init `components.json` on `userData` object

**Pull request merged**